StardewArchipelago/Archipelago/Gifting/GiftReceiver.cs

```diff
@@ -13,23 +13,25 @@ public class GiftReceiver
         private IMonitor _monitor;
         private ArchipelagoClient _archipelago;
         private IGiftingService _giftService;
+        private IGiftHandler _giftHandler;
         private StardewItemManager _itemManager;
         private Mailman _mail;
         private GiftProcessor _giftProcessor;
 
-        public GiftReceiver(IMonitor monitor, ArchipelagoClient archipelago, IGiftingService giftService, StardewItemManager itemManager, Mailman mail)
+        public GiftReceiver(IMonitor monitor, ArchipelagoClient archipelago, IGiftingService giftService,IGiftHandler giftHandler, StardewItemManager itemManager, Mailman mail)
         {
             _monitor = monitor;
             _archipelago = archipelago;
             _giftService = giftService;
+            _giftHandler = giftHandler;
             _itemManager = itemManager;
             _mail = mail;
             _giftProcessor = new GiftProcessor(monitor, archipelago, itemManager);
         }
 
         public void ReceiveAllGifts()
         {
-            var gifts = _giftService.GetAllGiftsAndEmptyGiftbox();
+            var gifts = _giftService.CheckGiftBox();

```

Comment:
Not emptying the whole giftbox at once will have a noticeable impact on performance, which makes me uneasy of this entire PR due to it.

I'm not sure what the correct answer is. I'm thinking, maybe track the ids you want to remove, then do one "RemoveMany(ids)" call at the end. Maybe make this whole thing async. Not sure.

But I don't think the way this is here is good enough. I could be convinced otherwise if you had some benchmarks that showed it's not noticeably slower, I didn't actively test it, it just seems like it'll be worse, and DataStorage operations are already one of the slowest things in the mod